app/main.py

```diff
@@ -1,8 +1,29 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people: list) -> list[Person]:
+    p_list = []
+
+    for person_data in people:
+        person = Person(person_data["name"], person_data["age"])
+        p_list.append(person)
+
+    for person_data, person in zip(people, p_list):
+        if "wife" in person_data and person_data["wife"] is not None:
+            wife_name = person_data["wife"]
+            wife = next((p for p in p_list if p.name == wife_name), None)
```

Comment:
You get wife object from Person.people by name
```suggestion
            wife = Person.people.get(wife_name)
```